src/software/ai/hl/stp/tactic/passer_tactic.cpp

```diff
@@ -85,12 +85,10 @@ void PasserTactic::calculateNextAction(ActionCoroutine::push_type& yield)
 
         // We want to keep trying to kick until the ball is moving along the pass
         // vector with sufficient velocity
-        Angle passer_to_receiver_angle =
-            (pass.receiverPoint() - pass.passerPoint()).orientation();
-        ball_velocity_to_pass_orientation =
-            ball.velocity().orientation().minDiff(passer_to_receiver_angle);
-    } while (ball_velocity_to_pass_orientation.abs() > Angle::fromDegrees(20) ||
-             ball.velocity().length() < 0.5);
+        Angle kick_direction = (pass.receiverPoint() - ball.position()).orientation();
+        hasKickedBall        = hasBallBeenKicked(ball, kick_direction);
+
+    } while (!hasKickedBall);
```

Comment:
just wondering why we need to store `hasKickedBall`, why not
```suggestion
         } while (!hasBallBeenKicked(ball, kick_direction));
```

Reply:
Thought it is better to keep `kick_direction` in the loop, whereas in this case we need to define it outside. It is more efficient though so I changed it to the suggested.